stereo_image_proc/launch/stereo_image_proc.launch.py

```diff
@@ -45,19 +45,30 @@ def generate_launch_description():
             description='Whether to use approximate synchronization of topics. Set to true if '
                         'the left and right cameras do not produce exactly synced timestamps.'
         ),
+        DeclareLaunchArgument(
+            name='image_sub_reliability', default_value='best_effort',
```

Comment:
How do you feel about the parameter name being `use_system_default_qos` of type bool and the description says that when false, uses sensor data qos? Else, uses system default QoS.

Reply:
This can work, though I'll point out it's possible that the system default has the same settings as SensorDataQoS since it just means we use whatever is configured in the underlying RMW. It means if users want a "reliable" pipeline, they should set `use_system_default_qos` to `true` and ensure their RMW is configured appropriately.

That being said, I'm fine changing the parameter for conciseness as it works for me.

Reply:
to clarify that: `use system default = true` I think is reliable for all RMWs, is that incorrect? Then `use system default = false` (default settings) I would expect to use sensor data qos. The parameter description then  can be more clear about the use of default vs sensor unreliable connections.

Reply:
> use system default = true I think is reliable for all RMWs, is that incorrect?

Perhaps it is reliable for the default configuration of RMWs shipped with ROS 2 at the moment, but I don't think it is always true. The setting can change outside the control of the stereo_image_proc nodes, which I think is fine since presumably the integrator is configuring the RMW.

According to the [design doc](http://design.ros2.org/articles/qos.html):

> To honor the QoS settings of the system, developers can use the rmw_qos_profile_system_default QoS profile which delegates the responsibility of the QoS machinery to the underlying DDS vendor. This allows developers to deploy ROS 2 applications and use DDS vendor tools to configure the QoS settings.

Reply:
If we go this route, we should just avoid giving the incorrect notion that the QoS will be reliable if `rmw_qos_profile_system_default` is used.

Reply:
OK, well I wouldn't necessarily expect that to change if that's how it is now (default behavior very rarely changes in large things like that). I know the Cyclone and Fast RTPS both have reliable as default from the users of the ouster drivers who requested that be added. 

I'm OK with another profile or system default that has an explicit call for reliable. I think the latter would  resolve our differences.

Reply:
I think changing the option to be "use system default" is good (it makes the node more configurable than just exposing the reliability setting). I think while we're at it, we should also toggle the profiles for the publishers as well.

Reply:
Sure, makes sense

Reply:
Updated, PTAL aea2740